File: blind-75/array/1-sum-of-two.py
class Solution(object):
    def twoSum(self, nums, target):
        visited = {}
        for i in range(len(nums)):
            pairValue = target - nums[i]
            if pairValue in visited:
                pairIndex = visited[pairValue]
                return [pairIndex, i]
            visited[nums[i]] = i
        return [-1, -1]

def main():
    solution = Solution()
    inputs = [[2,7,11,15], [2,9,4,11,6,7]]
    targets = [9, 16]
    for i in range(len(targets)):
        print(solution.twoSum(inputs[i], targets[i]))

main()

File: blind-JS45/array/1-SumOfTwo.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  // use Map initializer, instead of m = {}, it initializes as an object
  const visited = new Map();
  for (let i = 0; i < nums.length; i++) {
    pairValue = target - nums[i];
    // has function to check if map contains that key
    if (visited.has(pairValue)) {
      // get value by key
      pairIndex = visited.get(pairValue);
      return [pairIndex, i];
    }
    // set key value
    visited.set(nums[i], i);
  }
  return [-1, -1];
};
